angular/projects/admin/src/app/admin/media/media-grid/media-grid.component.ts

```diff
@@ -45,15 +44,21 @@ export class MediaGridComponent {
   }
 
   showDetails(file: TanamFile) {
-    this.dialog.open(MediaDialogDetailComponent, {
-      data: file
+    this.dialog.open(DialogComponent, {
+      data: {
+        ...file,
+        type: 'detail-file'
+      },
     });
     console.log(`[MediaGridComponent:showDetails] ${JSON.stringify({ file })}`);
   }
 
   remove(file: TanamFile) {
-    const dialogRef = this.dialog.open(MediaDialogDeleteComponent, {
-      data: file,
+    const dialogRef = this.dialog.open(DialogComponent, {
```

Comment:
See if you can abstract the way that the dialog is opened/handled so that it becomes a more easy to use interface for developers.

```typescript
const dialogRef = this.dialog.open(DialogComponent, DialogComponent.getConfig({
  title: "Good things to come",
  message: "It's never been a better time to be alive. Do you agree!?",
  buttons: ['yes', 'no'],
  icon: 'info',
}));
```

The `afterClose` stream returns the button value as defined in comment above (` 'ok' | 'yes' | 'no' | 'cancel'`)

Reply:
Resolve on this commit: https://github.com/oddbit/tanam/pull/115/commits/83e5b2f1baeae0016458c54f7094d881084c27cc